### PR TITLE
Init project

### PR DESCRIPTION
**Pull request by erkanerol**
***How does it work?***
Please check the README file in the PR.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by erkanerol** [comment]
@bavarianbidi can you please ping me when you are available so that we can go through some basic decisions in the operator?

File: controllers/rule_controller.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	dependents.SetGroupVersionKind(r.DeletionBlockRule.Dependent.GetSchemaGroupVersionKind())
+	if err := r.Client.List(ctx, dependents, &client.ListOptions{Namespace: managed.GetNamespace()}); err != nil {
+		return reconcile.Result{}, microerror.Mask(IgnoreNotFound(err))
+	}
+
+	allowed, err := r.DeletionBlockRule.CheckIsDeletionAllowed(*managed, *dependents)
+	if err != nil {
+		return reconcile.Result{}, microerror.Mask(IgnoreNotFound(err))
+	}
+	logger.Info("Deletion is", "allowed:", allowed)
```
Comment (bavarianbidi) [inline controllers/rule_controller.go:84]:
```suggestion
```

File: controllers/rule_controller.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	if allowed {
+		logger.Info("We can delete the finalizer. Removing finalizer")
+		controllerutil.RemoveFinalizer(managed, r.Finalizer)
+		if err := r.Client.Update(ctx, managed); err != nil {
+			return reconcile.Result{}, microerror.Mask(err)
+		}
+		return ctrl.Result{}, nil
+	} else {
+		return ctrl.Result{RequeueAfter: time.Minute * 1}, nil
```
Comment (bavarianbidi) [inline controllers/rule_controller.go:93]:
```suggestion
		logger.Info("Deletion is not allowed.")
		return ctrl.Result{RequeueAfter: time.Minute * 1}, nil
```

**Review by bavarianbidi — COMMENTED** [review]
really nice, like it.

as you already defined a metrics-port, would be great to see some metrics exposed. And it would be cool to have some tests in place.

I'm very interested to add this (metrics + tests) as a separate PR by my own .

**Pull request merged**